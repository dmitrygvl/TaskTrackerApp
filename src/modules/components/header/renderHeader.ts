// import "./renderHeader.scss";

export const renderHeader = (headerRoot: HTMLElement): void => {
  headerRoot.innerHTML = `
  <h1>Task Tracker app</h1>
  <nav class="'nav">
  <h1>My Task Tracker App</h1>
  <nav class="nav">
    <a href="/calendar" class="nav__link" id="calendar-link">Calendar</a>
    <a href="/list" class="nav__link" id="list-link">List</a>
    <a href="/about" class="nav__link" id="about-link">About</a>
  </nav>
  `;
};

// <nav>
//     <a href="/calendar" class="nav__link" id="">Tasks calendar</a>
//     <a href="/tasklist" class="nav__link" id="My taskList"></a>
//   </nav>
