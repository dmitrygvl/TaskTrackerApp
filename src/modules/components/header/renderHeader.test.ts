import { renderHeader } from "./renderHeader";

// describe('render header', () => {
//   beforeEach(() => {
//     const root = document.createElement('header');
//   });
//   afterEach(() => {
//     root?.remove();
//   });

//   it('Title is renders', () => {

//     renderHeader(root);
//     const titleElement = root.querySelector('.h1') as HTMLElement;
//     expect(titleElement).toBe('Task Tracker app');
//   })
//   it('Navigation in header is renders', () => {
//     const root = document.createElement('header');

//     renderHeader(root);
//     const navElement = root.querySelector('.nav') as HTMLElement;
//     const navLinks = navElement.querySelectorAll('.nav__link') as NodeListOf<HTMLAnchorElement>;
//     expect(navLinks.length).toBe(3);

//   })

// })
// // <h1>Task Tracker app</h1>

describe("renderHeader", () => {
  it("renders header with specific content", () => {
    const headerRoot = document.createElement("div");
    renderHeader(headerRoot);

    expect(headerRoot.innerHTML).toContain("Task Tracker app");
    expect(headerRoot.innerHTML).toContain("Calendar");
    expect(headerRoot.innerHTML).toContain("Task list");
    expect(headerRoot.innerHTML).toContain("About");
  });

  it("renders header with specific links", () => {
    const headerRoot = document.createElement("div");

    renderHeader(headerRoot);

    const links = headerRoot.querySelectorAll(".nav__link");
    expect(links.length).toBe(3);
    expect((links[0] as HTMLAnchorElement).href).toContain("/calendar");
    expect((links[1] as HTMLAnchorElement).href).toContain("/list");
    expect((links[2] as HTMLAnchorElement).href).toContain("/about");
  });

  it("renders header with specific IDs for links", () => {
    const headerRoot = document.createElement("div");

    renderHeader(headerRoot);

    const calendarLink = headerRoot.querySelector("#calendar-link");
    const listLink = headerRoot.querySelector("#list-link");
    const aboutLink = headerRoot.querySelector("#about-link");
    expect(calendarLink).toBeTruthy();
    expect(listLink).toBeTruthy();
    expect(aboutLink).toBeTruthy();
  });

  it("renders header using specific HTML structure", () => {
    const headerRoot = document.createElement("div");

    renderHeader(headerRoot);

    expect(headerRoot.innerHTML).toMatchSnapshot();
  });
});
