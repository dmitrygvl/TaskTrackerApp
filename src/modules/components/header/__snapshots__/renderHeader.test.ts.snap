// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderHeader renders header using specific HTML structure 1`] = `
"
  <h1>Task Tracker app</h1>
  <nav class="nav">
    <a href="/calendar" class="nav__link" id="calendar-link">Calendar</a>
    <a href="/list" class="nav__link" id="list-link">Task list</a>
    <a href="/about" class="nav__link" id="about-link">About</a>
  </nav>
  "
`;
